File: entites/Lump.py
class Lump:
    def __init__(self, filepos: int, size: int, name: str):
        self.filepos = filepos
        self.size = size
        self.name = name

    def __str__(self):
        return f"{self.filepos=}, {self.size=}, {self.name=}"

    __repr__ = __str__

File: cpp_code/entities/Lump.cpp
#include "../headers/Lump.h"
#include <cstdint>
#include <iostream>

typedef struct Lump  {
    uint16_t filepos;
    uint16_t size;
    char* name;

    // Constructor
    Lump(uint16_t filepos, uint16_t size, char* name)
        : filepos(filepos), size(size), name(name)
    {}

    friend std::ostream& operator<<(std::ostream& os, const Lump& obj) {
        os << "Lump{ ";
        os << "filepos: " << obj.filepos << " ";
        os << "size: " << obj.size << " ";
        os << "}";
        return os;
    }
} Lump;
